### PR TITLE
Opcode GASLIMIT implementation

### LINKED ISSUE #51: Opcode: GASLIMIT

### PR DESCRIPTION
**Pull request by JulianVentura**
Closes #51 
This PR implements the GASLIMIT opcode with its tests.

### CODE REVIEW

File: src/syscall.rs [hunk 1]
```diff
@@ -187,6 +187,12 @@ impl<'c> SyscallContext<'c> {
         value.hi = (aux >> 128).low_u128();
     }
 
+    #[allow(improper_ctypes)]
```
Comment (FrancoGiachetta) [inline src/syscall.rs:190]: This line shouldn't be needed anymore.

File: src/syscall.rs [hunk 2]
```diff
@@ -187,6 +187,12 @@ impl<'c> SyscallContext<'c> {
...
+    pub extern "C" fn store_in_gaslimit_ptr(&self, value: &mut U256) {
+        value.lo = self.env.tx.gas_limit as u128;
```
Comment (FrancoGiachetta) [inline src/syscall.rs:192]: Do we need to represent value as U256? Couldn't it be u64?

**Review by FrancoGiachetta — CHANGES_REQUESTED** [review]

File: src/syscall.rs [hunk 3]
```diff
@@ -345,6 +350,10 @@ pub fn register_syscalls(engine: &ExecutionEngine) {
             symbols::STORE_IN_CALLVALUE_PTR,
             SyscallContext::store_in_callvalue_ptr as *const fn(*mut c_void, *mut U256) as *mut (),
         );
+        engine.register_symbol(
+            symbols::GET_GASLIMIT,
+            SyscallContext::get_gaslimit as *const fn(*mut c_void, *mut U256) as *mut (),
```
Comment (FrancoGiachetta) [inline src/syscall.rs:355]: I think this `*mut U256` is not need any longer since the syscall is not using it now.

**Review by FrancoGiachetta — CHANGES_REQUESTED** [review]

**Review by FrancoGiachetta — APPROVED** [review]

**Comment by FrancoGiachetta** [comment]
LGTM
@MegaRedHand

File: tests/evm.rs [hunk 4]
```diff
@@ -182,6 +173,7 @@ fn test_opcode_origin() {
     let mut env = Env::default();
     let caller = Address::from_str("0x9bbfed6889322e016e0a02ee459d306fc19545d8").unwrap();
     env.tx.caller = caller;
+    env.tx.gas_limit = 999_999;
```
Comment (MegaRedHand) [inline tests/evm.rs:176]: Why is this needed?

File: tests/evm.rs [hunk 5]
```diff
@@ -744,7 +737,35 @@ fn callvalue_gas_check() {
 fn callvalue_stack_overflow() {
     let mut program = vec![Operation::Push0; 1024];
     program.push(Operation::Callvalue);
+    let mut env = Env::default();
+    env.tx.gas_limit = 999_999;
```
Comment (MegaRedHand) [inline tests/evm.rs:741]: Is this needed?

**Review by MegaRedHand — COMMENTED** [review]
Please check the setting of `env.tx.gas_limit`. I think most of them are OK with the default

**Review by MegaRedHand — APPROVED** [review]
LGTM

**Pull request merged**